apps/prairielearn/src/models/variant.ts

```diff
@@ -213,24 +215,30 @@ export async function selectAndAuthzVariant(options: {
     ) {
       const authnUserPermissions = await callRow(
         'authz_course_instance',
-        [authn_user.user_id, variant.course_instance_id, is_administrator, new Date(), null],
-        z.object({ has_course_instance_permission_view: z.boolean() }),
+        [authn_user.user_id, variant.course_instance_id, new Date()],
+        z.object({ course_instance_role: EnumCourseInstanceRoleSchema }),
       );
 
       const userPermissions = await callRow(
         'authz_course_instance',
-        [user.user_id, variant.course_instance_id, is_administrator, new Date(), null],
-        z.object({ has_course_instance_permission_view: z.boolean() }),
+        [user.user_id, variant.course_instance_id, new Date()],
+        z.object({ course_instance_role: EnumCourseInstanceRoleSchema }),
       );
 
-      authnHasCourseInstancePermissionView =
-        authnUserPermissions.has_course_instance_permission_view;
-      hasCourseInstancePermissionView = userPermissions.has_course_instance_permission_view;
+      authnHasCourseInstancePermissionView = calculateCourseInstanceRolePermissions(
+        authnUserPermissions.course_instance_role,
+      ).has_course_instance_permission_view;
+      hasCourseInstancePermissionView = calculateCourseInstanceRolePermissions(
+        userPermissions.course_instance_role,
+      ).has_course_instance_permission_view;
```

Comment:
Aside: as long as we're making changes to this, how about an API that's like `hasCoursePermission(course_instance_role, 'View')`? Could do the same for course instances.

Reply:
I don't find that API significantly cleaner. Maybe `isAtLeastRole(course_instance_role, "Viewer")`.

Reply:
You might be thinking about roles differently? We always think of roles as including the roles below them - that is, and `Editor` is a `Viewer` and does have `Viewer` permissions. We almost never care about exact equality for roles.

Reply:
I just don't want to invent another term for the permission: then we have "View", "Viewer", and the actual boolean permission.

Reply:
Maybe we can just reuse "hasRole"?

Reply:
Sorry, that was literally just a typo 🙃 I meant to suggest `hasCoursePermission(course_instance_role, 'Viewer')`

Reply:
I made a variant of this change in 67fccab046158363b1b7374d36a6ce80c8aa1ccb. It now looks like:

```
hasCourseInstancePermissionView = hasInstanceRole(
        calculateCourseInstanceRolePermissions(userPermissions.course_instance_role),
        'Student Data Viewer',
      );
```